decide/administration/views.py

```diff
@@ -20,6 +25,115 @@ def index(request):
+            if id_users is None:
+                users = User.objects.all()
+                id_users = [user.id for user in users]
+            if len(id_users) > 0:
+                for voter_id in id_users:
+                    census = Census(voting_id=voting_id, voter_id=voter_id)
+                    census.save()
+            return Response({"id": voting_id, "name": voting.name}, status=HTTP_200_OK)
+
+    def put(self, request):
+
+        votings_id = request.data.get("idList")
+        action = request.data.get('action')
+        if not action:
+            return Response({}, status=status.HTTP_400_BAD_REQUEST)
+        msg = ''
+        st = status.HTTP_200_OK
+        if action == 'start':
+            votings = Voting.objects.filter(id__in=votings_id, start_date__isnull=True)
+            if len(votings) > 0:
+                for voting in votings:
+                    voting.create_pubkey()
+                    voting.start_date = timezone.now()
+                    voting.save()
+                msg = 'Votings started'
+            else:
+                msg = 'All votings all already started'
+                st = status.HTTP_400_BAD_REQUEST
+
+        elif action == 'stop':
+            votings = Voting.objects.filter(id__in=votings_id, start_date__isnull=False, end_date__isnull=True)
+            if len(votings) > 0:
+                for voting in votings:
+                    voting.end_date = timezone.now()
+                    voting.save()
+                msg = 'Votings stopped'
+            else:
+                msg = 'All votings all already stopped or not started'
+                st = status.HTTP_400_BAD_REQUEST
+        elif action == 'tally':
+            votings = Voting.objects.filter(id__in=votings_id, start_date__isnull=False, end_date__isnull=False)
+            if len(votings) > 0:
+                for voting in votings:
+                    key = request.COOKIES.get('token', "")
+                    voting.tally_votes(key)
+                    msg = 'Votings tallied'
+            else:
+                msg = 'All votings all already tallied, not stopped or not started'
+                st = status.HTTP_400_BAD_REQUEST
+        else:
+            msg = 'Action not found, try with start, stop or tally'
+            st = status.HTTP_400_BAD_REQUEST
+        return Response(msg, status=st)
+
+    def delete(self, request):
+        if request.data.get("idList") is None:
+            Voting.objects.all().delete()
+            return Response({}, status=HTTP_200_OK)
+        else:
+            ids = request.data.get("idList")
+            is_valid(len(ids) > 0, 'The format of the ids list is not correct')
+            Voting.objects.filter(id__in=ids).delete()
+            return Response({}, status=HTTP_200_OK)
+
+class VotingsAPI(APIView):
+    permission_classes = (IsAdminAPI,)
+
+    def get(self, request, voting_id):
+        try:
+            query = Voting.objects.filter(id=voting_id).get()
```

Comment:
why is this called query?, maybe **_voting_** or **_result_**

Reply:
Query is also fine like it is used in djiango wiki for db operations but for readability has been changed in last commit.